### PR TITLE
Code improvements

### PR DESCRIPTION
**Pull request by MagdaPaj**
## Description

* Change all `g_command` variables to `constexpr const char*`. This change is made for consistency (we had some commands with `static const std::string` and some with `constexpr const char*`). Also the `g_command` variables are used in `RunCommand` method, which requires `char*` and not `std::string` type.
* Improve order of variables
* Use `const char*` for `sourcesDir` argument to make it consistent with the rest of the code (`sourcesDir` is part of the interface `PackageMangerConfigurationBase`. In other parts of the code (like Mmi interface, MmiSet, MmiGet) `const char*` is used instead of `std::string`).
* Fix TrimStart and TrimEnd and use the same code as in the HostNameBase.cpp. TrimStart and TrimEnd search the first or the last position of a character that doesn't match any of the trim characters. If our string doesn't contain any of the trim characters, then position 0 or the last position will be returned and the whole string will be correctly returned. `pos=npos` is returned only when all `str` characters match the `trim` characters, which means that the whole `str` needs to be trimmed and the returned value should be empty, and not the original string.

## Checklist

- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-osconfig/blob/main/CONTRIBUTING.md).
- [ ] I added unit-tests to validate my changes. All unit tests are passing.
- [ ] I have merged the latest `main` branch prior to this PR submission.
- [ ] I submitted this PR against the `main` branch.

### CODE REVIEW

**Review by Ioana37 — APPROVED** [review]
I think it looks cleaner like this :)

File: src/modules/packagemanagerconfiguration/src/lib/PackageManagerConfigurationBase.cpp [hunk 1]
```diff
@@ -586,7 +586,7 @@ std::string PackageManagerConfigurationBase::TrimStart(const std::string& str, c
     size_t pos = str.find_first_not_of(trim);
     if (pos == std::string::npos)
     {
-        return str;
+        return "";
```
Comment (Salazander) [inline src/modules/packagemanagerconfiguration/src/lib/PackageManagerConfigurationBase.cpp:589]: Isn't this change breaking 
 std::string packageElement = packageName + "=" + Trim(version, " "); ?
 
 Assumer version="1.0.1", wouldn't the new Trim-function result in "packageName=" whereas before it did result in "packageName=1.0.1" ?

Comment (MagdaPaj) [reply src/modules/packagemanagerconfiguration/src/lib/PackageManagerConfigurationBase.cpp:589]: Nope, it won't break. See my desription:
"TrimStart and TrimEnd search the first or the last position of a character that **doesn't** match any of the trim characters. If our string doesn't contain any of the trim characters, then position 0 or the last position will be returned and the whole string will be correctly returned. pos=npos is returned only when all str characters match the trim characters, which means that the whole str needs to be trimmed and the returned value should be empty, and not the original string."

Comment (Salazander) [reply src/modules/packagemanagerconfiguration/src/lib/PackageManagerConfigurationBase.cpp:589]: I missed the **_not_of** part... all good!

**Review by Salazander — APPROVED** [review]

**Pull request merged**